homeassistant/components/switchbot_cloud/entity.py

```diff
@@ -0,0 +1,51 @@
+"""Base class for SwitchBot via API entities."""
+import logging
+from typing import Any
+
+from switchbot_api import Device, Remote, SwitchBotAPI
+
+from homeassistant.helpers.update_coordinator import (
+    CoordinatorEntity,
+    DataUpdateCoordinator,
+)
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+
+class SwitchBotCloudEntity(CoordinatorEntity):
+    """Representation of a SwitchBot Cloud entity."""
+
+    _api: SwitchBotAPI
+    _is_remote = False
+    _SwitchBot_state: dict[str, Any] | None = None
+
+    def __init__(
+        self,
+        api: SwitchBotAPI,
+        device: Device | Remote,
+        coordinator: DataUpdateCoordinator,
+    ) -> None:
+        """Initialize the entity."""
+        super().__init__(coordinator=coordinator)
+        self._api = api
+        self._attr_unique_id = device.device_id
+        self._attr_name = device.device_name
+        self._is_remote = isinstance(device, Remote)
```

Comment:
This is leaking the abstraction into `entity.py`. This base class shouldn't know about `Remote`

Reply:
Create a new subclass in `switch.py` for remotes or set it there.